stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -574,6 +574,12 @@ function _factorize(A::HermOrSym{T}; check::Bool=true) where T
     end
 end
 
+logabsdet(A::RealHermSymComplexHerm) = logabsdet(_factorize(A; check=false))
+logabsdet(A::Symmetric{<:Real}) = logabsdet(_factorize(A; check=false))
+logabsdet(A::Symmetric) = logabsdet(_factorize(A; check=false))
+logdet(A::RealHermSymComplexHerm) = real(logdet(_factorize(A; check=false)))
+logdet(A::Symmetric{<:Real}) = logdet(_factorize(A; check=false))
+logdet(A::Symmetric) = logdet(_factorize(A; check=false))
```

Comment:
`Symmetric{<:Real}` is included in `RealHermSymComplexHerm`, so if you want the same definition to work, then there's no need to have that extra method.

```suggestion
logabsdet(A::RealHermSymComplexHerm) = logabsdet(_factorize(A; check=false))
logabsdet(A::Symmetric) = logabsdet(_factorize(A; check=false))
logdet(A::RealHermSymComplexHerm) = real(logdet(_factorize(A; check=false)))
logdet(A::Symmetric) = logdet(_factorize(A; check=false))
```

Reply:
It looks like removing these definitions has introduced a method signature ambiguity:
```
  MethodError: logdet(::LinearAlgebra.Symmetric{Float32, Matrix{Float32}}) is ambiguous.
  Candidates:
    logdet(A::LinearAlgebra.Symmetric)
      @ LinearAlgebra /cache/build/default-amdci4-7/julialang/julia-master/julia-e9c7325a82/share/julia/stdlib/v1.11/LinearAlgebra/src/symmetric.jl:580
    logdet(A::Union{LinearAlgebra.Hermitian{T, S}, LinearAlgebra.Hermitian{Complex{T}, S}, LinearAlgebra.Symmetric{T, S}} where {T<:Real, S})
      @ LinearAlgebra /cache/build/default-amdci4-7/julialang/julia-master/julia-e9c7325a82/share/julia/stdlib/v1.11/LinearAlgebra/src/symmetric.jl:579
  Possible fix, define
    logdet(::LinearAlgebra.Symmetric{T, S} where {T<:Real, S<:(AbstractMatrix{<:T})})

```
Going to add them back in.